### PR TITLE
Add CLion plugin

### PR DESCRIPTION
**Pull request by vol0n**
How to launch plugin. 

1. Download IntelliJ IDEA ide
2. git clone https://github.com/vol0n/UTBotCpp.git
3. git checkout addClionPlugin
4. open the project in ide
5. Run gradle task 'Run plugin`
6. Configure the plugin: all steps are in README.md of this pr

I copied it here: 

1. Open plugin settings in settings - Tools - UTBot Settings
2. Click `detect paths`. It will try to get source paths, build dir paths from CLion 
CMake model.
3. Specify absolute path to build folder, it should be different from build folder that CLion uses, 
because there can be conflicts between UTBotCpp and CLion. For example, if CLion uses `project_path/cmake-build-debug`, 
then you can specify `project_path/utbot_build`.
4. For target path specify `/utbot/auto/target/path`
5. Specify path to test folder.
6. Specify name of the server and port. 
8. If you use docker to run UTBotCpp and your project is mounted to docker, 
you can specify path to project inside docker and the plugin should work.

### CODE REVIEW

File: clion-plugin/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# IntelliJ Platform Artifacts Repositories
+# -> https://plugins.jetbrains.com/docs/intellij/intellij-artifacts.html
+
+pluginGroup = com.github.vol0n.utbotcppclion
```
Comment (korifey) [inline clion-plugin/gradle.properties:4]: Plugin group must be changed to `com.huawei.utbot`

File: clion-plugin/src/main/kotlin/actions/GenerateForProjectAction.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package actions
```
Comment (korifey) [inline clion-plugin/src/main/kotlin/actions/GenerateForProjectAction.kt:1]: package names must be changed everywhere: `com.huawei.utbot.cpp`

File: clion-plugin/src/main/resources/logback.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    <appender name="DummyServerAppender" class="server.DummyServerAppender">
+        <encoder>
+            <pattern>%date{dd MMM yyyy HH:mm:ss} | %20.25file:%3.100line |%p| %m%n</pattern>
+        </encoder>
+    </appender>
+
+    <logger name="ClientLogger">
+        <appender-ref ref="ClientAppender" />
+        <level level="trace" />
```
Comment (korifey) [inline clion-plugin/src/main/resources/logback.xml:16]: mustn't be trace by default ?

File: clion-plugin/src/test/kotlin/com/github/vol0n/utbotcppclion/MyPluginTest.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.github.vol0n.utbotcppclion
+
```
Comment (korifey) [inline clion-plugin/src/test/kotlin/com/github/vol0n/utbotcppclion/MyPluginTest.kt:2]: what does it test?

**Comment by korifey** [comment]
@vol0n please description into this PR:
1. How to test plugin (exact steps for me to test is locally)
2. How to develop plugin (same as CONTRIBUTING.MD) in this repo
3. Test cases - all step-by-step scenarios with images how to use this plugin

File: clion-plugin/src/main/kotlin/server/GrpcServer.kt [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+        override fun generateFileTests(request: Testgen.FileRequest): Flow<Testgen.TestsResponse> {
+            logStarted(::generateFileTests)
+            val projectPath = request.projectRequest.projectContext.projectPath
+            val pathToGeneratedTestFile = Paths.get(
+                projectPath,
+                request.projectRequest.projectContext.testDirPath,
+                request.filePath
+            ).toString()
+            val generatedCode = "Hello " + File("${projectPath}/${request.filePath}").readText()
```
[the remark this answers is not in the document]
Comment (vol0n) [reply clion-plugin/src/main/kotlin/server/GrpcServer.kt:90]: Yes, It was used when UTBotCpp was not published, after publication it was not used. But this stub can be used for testing plugin without installing the server

File: clion-plugin/src/test/kotlin/com/github/vol0n/utbotcppclion/MyPluginTest.kt (hunk 4 above)
Comment (vol0n) [reply clion-plugin/src/test/kotlin/com/github/vol0n/utbotcppclion/MyPluginTest.kt:2]: It is a template for writing tests in IntelliJ Platform, so it tests nothing.  Right now I am working on actual tests

**Pull request closed without merging**